### ISSUE TITLE
2017.11.27.00 OS X 10.11 build failure: conversion function from 'folly::dynamic::ObjectMaker' to 'folly::dynamic' invokes a deleted function

### ISSUE THREAD

**Issue by ilovezfs** [facebook/folly#718]
CC @Orvid @wez

```
libtool: compile:  /usr/bin/clang++ -std=c++11 -stdlib=libc++ -DHAVE_CONFIG_H -I../../.. -pthread -I/usr/local/include -I/usr/local/Cellar/openssl/1.0.2m/include -I/usr/local/opt/openssl/include -F/usr/local/Frameworks -std=gnu++1y -Os -w -pipe -march=native -mmacosx-version-min=10.11 -c LogConfigParser.cpp  -fno-common -DPIC -o .libs/LogConfigParser.o
LogConfigParser.cpp:548:10: error: conversion function from 'folly::dynamic::ObjectMaker' to 'folly::dynamic' invokes a deleted function
  return value;
         ^~~~~
../../../folly/dynamic.h:164:18: note: 'dynamic' has been explicitly marked deleted here
  /* implicit */ dynamic(ObjectMaker const&) = delete;
                 ^
1 error generated.
make[4]: *** [LogConfigParser.lo] Error 1
make[3]: *** [all-recursive] Error 1
make[2]: *** [all-recursive] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```
Ref https://github.com/Homebrew/homebrew-core/pull/21091
https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/13723/
https://gist.github.com/ilovezfs/9dffb1c2b7a1294fa3a2f1ec6150a1b7

**Comment by wez** [comment]
cc: @simpkins

**Comment by simpkins** [comment]
This appears to be an issue with older C++ compilers.  Old versions of the C++ standard required an explicit `std::move()` around value here.  This was addressed by DR 1579, and g++ no longer requires this starting in version 5.x+
http://www.open-std.org/jtc1/sc22/wg21/docs/cwg_defects.html#1579

I'll land a fix for this.

**Comment by ilovezfs** [comment]
@simpkins looks like this was fixed in 2017.12.04.00, yes?

**Comment by simpkins** [comment]
Yes, this was fixed by commit ad993b99e32
